website/scripts/ros/ros_erv_status.js

```diff
@@ -0,0 +1,89 @@
+/***
+ * Subscribe to approaching_erv_status ROS2 topic and display the emerency response vehicle status information
+ * Command to publish messages:
+ *
+ros2 topic pub -r  0.5 /guidance/approaching_erv_status carma_msgs/msg/UIInstructions 'stamp:
+  sec: 0
+  nanosec: 0
+msg: 'HAS_APPROACHING_ERV:0'
+type: 0
+response_service: '''
+
+ros2 topic pub -r  0.5 /guidance/approaching_e^Crv_status carma_msgs/msg/UIInstructions 'stamp:
+  sec: 0
+  nanosec: 0
+msg: 'HAS_APPROACHING_ERV:1,TIME_UNTIL_PASSING:11.6,EGO_VEHICLE_ACTION:Changing\ lanes\ to\ the\ left.'
+type: 0
+response_service: '''
+ */
+const HAS_APPROACHING_ERV = "HAS_APPROACHING_ERV";
+const TIME_UNTIL_PASSING = "TIME_UNTIL_PASSING";
+const EGO_VEHICLE_ACTION = "EGO_VEHICLE_ACTION";
+/***
+ * Subscribe ERV approaching status ROS2 topics and display the warning message on the UI if ERV is detected.
+ * If not ERV detected, hide the warning message.
+ */
+function subscribeToERVStatusInfo() {
+    console.log("subscribeToERVStatusInfo")
+    let imgPath = "../../images/ambulance-orange.svg";
+    let listener = new ROSLIB.Topic({
+        ros: g_ros,
+        name: T_ERV_STATUS,
+        messageType: M_ERV_UIInstruction
+    });
+
+    listener.subscribe(function (message) {
+        if (message.msg !== undefined) {
+            let statusMsg = message.msg.split(",");
+            let statusMap = {};
+            if (statusMsg.length >= 1) {
+                statusMap[statusMsg[0].split(":")[0]] = statusMsg[0].split(":")[1];
+            }
+            if (statusMsg.length >= 2) {
+                statusMap[statusMsg[1].split(":")[0]] = statusMsg[1].split(":")[1];
+            }
+            if (statusMsg.length >= 3) {
+                statusMap[statusMsg[2].split(":")[0]] = statusMsg[2].split(":")[1];
+            }
+            if (statusMap[HAS_APPROACHING_ERV] === "1") {
+                updateERVStatusDivByEventInfo(imgPath, "<p>Approaching ERV detected!</p><p>Time until passing (secs):"
+                    + " <span style=\'color: rgb(167, 223, 57) !important\'>" +
+                    statusMap[TIME_UNTIL_PASSING] + "</p><p> Ego Vehicle Status:: " +
+                    statusMap[EGO_VEHICLE_ACTION] + "</p>", 'border-warning');
+                //Play sound to notify driver
+                playSound('audioAlert1', true);
```

Comment:
Could we update this to only play the sound when transitioning from `statusMap[HAS_APPROACHING_ERV] === "0"` to `statusMap[HAS_APPROACHING_ERV] === "1"`? Otherwise, since this message is published by carma-platform's approaching_emergency_vehicle_plugin at 1 Hz, the sound could occur every second until a detected ERV passes the host vehicle.

Reply:
https://github.com/usdot-fhwa-stol/carma-web-ui/pull/160/commits/330718d1074f5e0f00ced75dce8060424044ef86